test/e2e/common.go

```diff
@@ -359,3 +360,56 @@ func IsBootedFromDisk(client *ssh.Client) (bool, error) {
 
 	return isDisk, nil
 }
+
+type BuildAndInstallDeploymentInput struct {
+	Kustomization       string
+	ClusterProxy        framework.ClusterProxy
+	WaitForDeployment   bool
+	WatchDeploymentLogs bool
+	DeploymentName      string
+	DeployNS            string
+	LogPath             string
+	WaitIntervals       []interface{}
+}
+
+func BuildAndInstallDeployment(ctx context.Context, input *BuildAndInstallDeploymentInput) error {
+	var err error
+	kustomization := input.Kustomization
+	clusterProxy := input.ClusterProxy
+	manifest, err := buildKustomizeManifest(kustomization)
+	if err != nil {
+		return err
+	}
+	err = clusterProxy.Apply(ctx, manifest)
+	if err != nil {
+		return err
+	}
+
+	ironicBMODeployment := &v1.Deployment{
```

Comment:
```suggestion
	deployment := &v1.Deployment{
```

Does this work if no name or namespace is specified (if we do not ask for wait and watch)? If not, I would like to put it inside an if statement so it is only executed when really needed.